src/CMakeLists.txt

```diff
@@ -39,45 +49,33 @@ include(gcc_version.cmake)
 # Figure out how to build and where to install the Python modules
 #
 
-find_program(PYTHON_EXECUTABLE NAMES python2)
-
-if (APPLE)
-  # Ask Python to tell us it's include directory, if nobody else has
-  if(NOT DEFINED PYTHON_INCLUDE_DIRS)
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-      "from distutils.sysconfig import get_python_inc;print get_python_inc()"
-      OUTPUT_VARIABLE PYTHON_INCLUDE_DIRS
-      RESULT_VARIABLE PYTHON_INCLUDE_DIR_RESULT ERROR_QUIET)
-    if(NOT PYTHON_INCLUDE_DIR_RESULT MATCHES 0)
-      message(SEND_ERROR "Failed to determine PYTHON_INCLUDE_DIR")
-    endif(NOT PYTHON_INCLUDE_DIR_RESULT MATCHES 0)
-  endif(NOT DEFINED PYTHON_INCLUDE_DIRS)
-
-  # Similar for the Python library
-  if(NOT DEFINED PYTHON_LIBRARY)
-      # Get the library path
-      execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c
-        "from distutils import sysconfig;print sysconfig.get_config_var('LIBDIR')"
-        OUTPUT_VARIABLE PYTHON_LIBRARY_DIR
-        RESULT_VARIABLE PYTHON_LIBRARY_DIR_RESULT ERROR_QUIET)
-      string(STRIP ${PYTHON_LIBRARY_DIR} PYTHON_LIBRARY_DIR)
-      if(NOT PYTHON_LIBRARY_DIR_RESULT MATCHES 0)
-          message(SEND_ERROR "Failed to determine PYTHON_LIBRARY")
-      endif(NOT PYTHON_LIBRARY_DIR_RESULT MATCHES 0)
-
-      # Get library filename - might not be safe to assume .dylib extension?
-      execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c
-        "import sys;print 'libpython%d.%d.dylib'%sys.version_info[0:2]"
-        OUTPUT_VARIABLE PYTHON_LIBRARY_FILE
-        RESULT_VARIABLE PYTHON_LIBRARY_FILE_RESULT ERROR_QUIET)
-      string(STRIP ${PYTHON_LIBRARY_FILE} PYTHON_LIBRARY_FILE)
-      if(NOT PYTHON_LIBRARY_FILE_RESULT MATCHES 0)
-          message(SEND_ERROR "Failed to determine PYTHON_LIBRARY")
-      endif(NOT PYTHON_LIBRARY_FILE_RESULT MATCHES 0)
-
-      set(PYTHON_LIBRARY "${PYTHON_LIBRARY_DIR}/${PYTHON_LIBRARY_FILE}")
-  endif(NOT DEFINED PYTHON_LIBRARY)
-endif (APPLE)
+find_package(PythonInterp)
+
+if (PYTHONINTERP_FOUND)
+  if (UNIX AND NOT APPLE)
+    if (PYTHON_VERSION_MAJOR EQUAL 3)
+        find_package(Boost COMPONENTS python${PYTHON_VERSION_SUFFIX})
+        find_package(PythonInterp 3)
+        find_package(PythonLibs 3 REQUIRED)
+    else()
+        find_package(Boost COMPONENTS python)
+        find_package(PythonInterp)
+        find_package(PythonLibs REQUIRED)
+    endif()
+  else()	
+    if (PYTHON_VERSION_MAJOR EQUAL 3)
+        find_package(Boost COMPONENTS python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+        find_package(PythonInterp 3)
+        find_package(PythonLibs 3 REQUIRED)
+    else()
+        find_package(Boost COMPONENTS python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+        find_package(PythonInterp)
+        find_package(PythonLibs REQUIRED)
+    endif()
+  endif()
+else()
+    message("Python not found")
+endif()
```

Comment:
I copied the above from another library, because the homebrew-installed python cannot be found by using:
`find_package(Boost COMPONENTS python` on macOS
It needs a version postfix like: `find_package(Boost COMPONENTS python27)`

There is Python version 3 stuff in there which we should remove, or actually try out and see if we can support it